src/util/TraceUtil.v3

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 Wizard Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+component TraceUtil {
+    // Renders a right-aligned string in a column of length `len`
+    def renderRight(out: TraceBuilder, len: int, f: TraceBuilder -> void) {
```

Comment:
Today I added a utility method to `StringBuilder` (which `TraceBuilder` will then inherit): `rjustify`, which allows you to retro-actively insert spaces `N` characters back. I think you can make use of it here, or in the caller(s).